maa-cli/src/run/copilot.rs

```diff
@@ -0,0 +1,207 @@
+use std::{
+    fs::{self, File},
+    io::{self, BufReader, Write},
+};
+
+use super::{run, Result};
+use crate::{
+    config::task::{
+        default_variants, task_type::TaskType, value::input::Input, Strategy, Task, TaskList, Value,
+    },
+    dirs::Dirs,
+    installer::maa_core::find_resource,
+    object,
+};
+use anyhow::Context;
+use reqwest::blocking::Client;
+use serde_json::Value as JsonValue;
+
+pub fn copilot(dirs: &Dirs, link: String) -> bool {
+    let api = "https://prts.maa.plus/copilot/get/";
+    let cache_dir = dirs.cache().display().to_string();
+    let base_resource_dir = find_resource(dirs).context("Failed to find resource!");
+    let stage_dir = format!(
+        "{}/Arknights-Tile-Pos",
+        base_resource_dir.unwrap().to_string_lossy()
+    );
+    let copilot_id = match link.get(link.len() - 5..) {
+        Some(id) => id,
+        None => {
+            println!("Invaild maa code.");
+            return false;
+        }
+    };
+    let url = format!("{}{}", api, copilot_id);
+
+    // Visit the API to obtain the data of homework.
+    let client = Client::new();
+    let response = client.get(&url).send().unwrap();
+    let json: JsonValue = response.json().unwrap();
+    let status_code = json["status_code"].clone().to_string();
+    if status_code != "200" {
+        println!("Request failed");
+        return false;
+    }
+
+    // Extract the response
+    let context = json["data"]["content"].clone();
+    let content = match context.as_str() {
+        Some(content) => content.to_string(),
+        None => {
+            println!("Error parsing JSON");
+            return false;
+        }
+    };
+
+    let recast: JsonValue = serde_json::from_str(&content).unwrap();
+
+    // Determine type of stage
+    let mut task_type: &str = "Copilot";
+    let content_type = recast["type"].clone();
+    if content_type == "SSS" {
+        task_type = "SSSCopilot";
+    }
+
+    // Print stage info
+    println!("Stage info:\n");
```

Comment:
The `println!` can be replaced `normal!` or `info!` provided by the log module. This way, we can control their output with `-v` and `-q` flags.

```suggestion
    info!("Stage info:");
```